src/charm.py

```diff
@@ -20,34 +21,75 @@ class ParcaScrapeTargetCharm(ops.CharmBase):
     def __init__(self, *args):
         super().__init__(*args)
 
+        # event handlers
+        self.framework.observe(self.on.collect_unit_status, self._on_collect_unit_status)
+
+        self._reconcile()
+
+    # EVENT HANDLERS
+    def _on_collect_unit_status(self, event: ops.CollectStatusEvent):
```

Comment:
I suggest we keep more or less the same structure we have in parca-k8s, for consistency.
at the top, reconcile
then, utility methods such as _scheme, external_hostname, etc..
then, config and relation management methods (used in reconcile or init)
and at the very end, event handlers and finally collect_unit_status 

I like that to be at the end because it's where things get wrapped up, and logically, the status depends on all that's above it.